operator/internal/status.go

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package operator
+
+import (
+	"context"
+	"time"
+
+	operatorapi "github.com/openperouter/openperouter/operator/api/v1alpha1"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ResourcesNotReadyError struct {
+	Message string
+}
+
+func (e ResourcesNotReadyError) Error() string { return e.Message }
+
+func (e ResourcesNotReadyError) Is(target error) bool {
+	_, ok := target.(*ResourcesNotReadyError)
+	return ok
+}
+
+type ResourcesUnexpectedError struct {
+	Message string
+}
+
+func (e ResourcesUnexpectedError) Error() string { return e.Message }
+
+func (e ResourcesUnexpectedError) Is(target error) bool {
+	_, ok := target.(*ResourcesUnexpectedError)
+	return ok
+}
+
+type DeploymentNotReadyError struct{}
+
+const (
+	ConditionAvailable   = "Available"
+	ConditionProgressing = "Progressing"
+	ConditionDegraded    = "Degraded"
+	ConditionUpgradeable = "Upgradeable"
+)
+
+func UpdateStatus(ctx context.Context, client k8sclient.Client, openperouter *operatorapi.OpenPERouter, condition string, reason string, message string) error {
```

Comment:
I'd skip updating the status for now. I don't think we ever relied on mtallb's one